libs/auth/testing/src/drivers/in-memory/register/register.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable, Inject } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { mergeMapTo } from 'rxjs/operators';
+
+import {
+  DaffLoginDriver,
+  DaffRegisterServiceInterface,
+  DaffLoginServiceInterface,
+  DaffRegisterRequest,
+  DaffRegisterResponse,
+  DaffLoginRequest,
+  DaffLoginResponse,
+  DaffCustomerRegistration
+} from '@daffodil/auth';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffInMemoryRegisterService implements DaffRegisterServiceInterface<
+  DaffRegisterRequest<DaffCustomerRegistration>,
+  DaffCustomerRegistration,
+  DaffRegisterResponse
+> {
+  url = '/api/auth/';
+
+  constructor(
+    private http: HttpClient,
+    @Inject(DaffLoginDriver) private loginDriver: DaffLoginServiceInterface<DaffLoginRequest, DaffLoginResponse>
+  ) {}
+
+  register(registration: DaffRegisterRequest<DaffCustomerRegistration>): Observable<DaffRegisterResponse> {
+    return this.http.post<any>(`${this.url}register`, registration).pipe(
```

Comment:
This violates the type constraint of the interface.

Reply:
The in-memory backend interface?

Reply:
Is this what I should do?
```ts
this.http.post<DaffCustomerRegistration>(`${this.url}register`, registration).pipe(
```

Reply:
Yes. That's better, but I believe you're returning a `DaffLoginResponse` at the moment.

Reply:
They are both aliases of `DaffAuthToken`, so its functionally the same. Would you prefer typing it as `DaffLoginResponse` or `DaffRegisterResponse`?

Reply:
I'd expect a `DaffRegisterResponse` for keeping the type the visibly consistent.